commit ccbd1e51911fb31c2f0063963b6591b586e4f83d
Author: Jeff Law <jeffreyalaw@gmail.com>
Date:   Fri Feb 18 09:11:13 2022 -0500

    Revert "tree: tweak warn_deprecated_use"
    
    This reverts commit 1b71bc7c8b18bd1b22debfde155f175fd1654942.

diff --git a/gcc/tree.cc b/gcc/tree.cc
index 2bbef2d6b75..dd919ff0717 100644
--- a/gcc/tree.cc
+++ b/gcc/tree.cc
@@ -12047,12 +12047,10 @@ warn_deprecated_use (tree node, tree attr)
 	attr = DECL_ATTRIBUTES (node);
       else if (TYPE_P (node))
 	{
-	  tree decl = TYPE_STUB_DECL (TYPE_MAIN_VARIANT (node));
+	  tree decl = TYPE_STUB_DECL (node);
 	  if (decl)
-	    {
-	      node = TREE_TYPE (decl);
-	      attr = TYPE_ATTRIBUTES (node);
-	    }
+	    attr = lookup_attribute ("deprecated",
+				     TYPE_ATTRIBUTES (TREE_TYPE (decl)));
 	}
     }
 
